pkg/dispatchcli/cmd/update_baseimage.go

```diff
@@ -0,0 +1,147 @@
+///////////////////////////////////////////////////////////////////////
+// Copyright (c) 2018 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+///////////////////////////////////////////////////////////////////////
+
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"strconv"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/vmware/dispatch/pkg/dispatchcli/i18n"
+	"github.com/vmware/dispatch/pkg/image-manager/gen/client/base_image"
+	"github.com/vmware/dispatch/pkg/image-manager/gen/models"
+)
+
+var (
+	updateBaseImageLong    = "Updates a base image from a given json representation"
+	updateBaseImageExample = `{
+		"dockerUrl": "vmware/dispatch-openfaas-nodejs6-base:0.0.3-dev1",
+		"groups": null,
+		"language": "nodejs6",
+		"name": "nodejs6-base",
+		"public": true,
+		"reason": null,
+		"tags": [
+			{
+				"key": "role",
+				"value": "test"
+			}
+		]
+	}`
+	updateFilePath          string
+	imageURLUpdateValue     []string
+	publicFlagUpdateValue   []string
+	languageFlagUpdateValue []string
+)
+
+func CallUpdateBaseImage(input interface{}) error {
+	baseImage := input.(*models.BaseImage)
+	params := base_image.NewUpdateBaseImageByNameParams()
+	params.BaseImageName = *baseImage.Name
+	params.Body = baseImage
+	_, err := imageManagerClient().BaseImage.UpdateBaseImageByName(params, GetAuthInfoWriter())
+
+	if err != nil {
+		return formatAPIError(err, params)
+	}
+
+	return nil
+}
+
+func NewCmdUpdateBaseImage(out io.Writer, errOut io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "base-image BASE_IMAGE_NAME [--base-image-file BASE_IMAGE_FILE] [--image-url IMAGE_URL] [--public] [--language LANGUAGE]",
+		Short:   i18n.T("Update base image"),
+		Long:    updateBaseImageLong,
+		Example: updateBaseImageExample,
+		Args:    cobra.MinimumNArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			err := updateBaseImage(out, errOut, cmd, args)
+			CheckErr(err)
+		},
+	}
+
+	cmd.Flags().StringVar(&updateFilePath, "base-image-file", "", "A json file to update the base image")
+	cmd.Flags().StringArrayVar(&imageURLUpdateValue, "image-url", []string{}, "The url for the container image.")
+	cmd.Flags().StringArrayVar(&publicFlagUpdateValue, "public", []string{}, "Whether the base image is public or private")
+	cmd.Flags().StringArrayVar(&languageFlagUpdateValue, "language", []string{}, "Specify the runtime language for the image")
+	return cmd
+}
+
+func updateBaseImage(out io.Writer, errOut io.Writer, cmd *cobra.Command, args []string) error {
+	baseImageName := args[0]
+	baseImage := models.BaseImage{}
+
+	if updateFilePath != "" && (len(imageURLUpdateValue) == 0 || len(publicFlagUpdateValue) == 0 || len(languageFlagUpdateValue) == 0) {
```

Comment:
Rather than all this logic, what happens if you just fetch the whole base image and simply overlay the updated fields?  Also, the file input format should not be the JSON output from the API.  Rather, we'd support a version similar to examples/seed.yaml, as we want to be consistent across the CLI.

Reply:
The yaml version is still supported via the command `dispatch update -f update.yaml`. If you think this json version is unnecessary I can remove it.

Reply:
ok, I was confused.  Yes, remove the JSON version.